metatensor-torch/include/metatensor/torch/labels.hpp

```diff
@@ -98,7 +99,7 @@ class METATENSOR_TORCH_EXPORT LabelsHolder: public torch::CustomClassHolder {
     TorchLabels rename(std::string old_name, std::string new_name) const;
 
     /// Move the values for these Labels to the given `device`
-    void to(torch::Device device);
+    TorchLabels to(torch::Device device) const;
```

Comment:
```suggestion
    TorchLabels to(torch::IValue device) const;
```

And then call `normalize_device` internally. This way we don't need an additional lambda function in `register.cpp`